hack/dev/Makefile

```diff
@@ -29,7 +29,7 @@ kubeconfig:
 .PHONY: clean
 clean: down
 	-@rm talosconfig
-	-@rm kubeconfig
+	-@rm -rf kubeconfig
```

Comment:
kubeconfig is a directory, as it's mounted into the volume (I'm not sure if it's what's expected, but this fixes `make clean` to pass without errors)

Reply:
Not expected, but this bug is annoying so we can keep this.